src/main/java/com/book/service/UserService.java

```diff
@@ -31,4 +31,28 @@ public Optional<UsersEntity> registerUser(UsersEntity usersEntity) {
         return usersEntityOptional;
 
     }
+
+    @Override
+    @Transactional
+    public Optional<UsersEntity> updateUser(UsersEntity usersEntity) {
+        Optional<UsersEntity> usersEntityOptional = userRepository.findUsersEntityByUserName(usersEntity.getUserName());
+        Optional<UsersEntity> result = Optional.empty();
+
+        if(usersEntityOptional.isPresent()){
+            String encryptedPassword = passwordEncoder.encode(usersEntity.getPassword());
+            usersEntityOptional.get().setPassword(encryptedPassword);
+            result = Optional.of(userRepository.save(usersEntityOptional.get()));
+        }
+
+        return result;
+    }
+
+    @Override
+    @Transactional
+    public Optional<UsersEntity> getUserByUserName(String userName) {
+        Optional<UsersEntity> usersEntityOptional = userRepository.findUsersEntityByUserName(userName);
+
+        return usersEntityOptional;
```

Comment:
This method only has 1 declaration and 1 return statement, 
so you can directly `return userRepository.findUsersEntityByUserName(userName);`